src/main.cpp

```diff
@@ -0,0 +1,235 @@
+    std::cout << segments << " " << segments.empty() << std::endl;
+    switch (parser.get().method()) {
+        case boost::beast::http::verb::get:
+            if (segments.empty() || params.contains("encoding-type") || params.contains("list-type")) {
+                // Among other things, listobjects should be handled here.
+
+                // This is a weird little thing because the parameters are a multimap.
+                auto f = url.params().find("list-type");
+
+                // Honestly not being able to use -> here strikes me as a potential mistake that
+                // will be corrected in the future when boost::url is released properly as part
+                // of boost
+                if (f != url.params().end() && (*f).value == "2") {
+                    co_await irods::s3::actions::handle_listobjects_v2(socket, parser, url);
+                }
+            }
+            else {
+                // GetObject
+                std::cout << "getobject detected" << std::endl;
+                co_await irods::s3::actions::handle_getobject(socket, parser, url);
+            }
+            break;
+        case boost::beast::http::verb::put:
+            if (parser.get().find("x-amz-copy-source") != parser.get().end()) {
+                // copyobject
+                std::cout << "Copyobject detected" << std::endl;
+                co_await irods::s3::actions::handle_copyobject(socket, parser, url);
+            }
+            else {
+                // putobject
+                std::cout << "putobject detected" << std::endl;
+                co_await irods::s3::actions::handle_putobject(socket, parser, url);
+                co_return;
+            }
+            break;
+        case boost::beast::http::verb::delete_:
+            if (segments.empty()) {
+                std::cout << "Deletebucket detected?" << std::endl;
+            }
+            else {
+                std::cout << "Deleteobject detected" << std::endl;
+                co_await irods::s3::actions::handle_deleteobject(socket, parser, url);
+                co_return;
+            }
+            break;
+        case boost::beast::http::verb::head:
+            // Probably just headbucket and headobject here.
+            // and headbucket isn't on the immediate list
+            // of endpoints.
+            if (url.segments().empty()) {
+                std::cout << "Headbucket detected" << std::endl;
+            }
+            else {
+                std::cout << "Headobject detected" << std::endl;
+                co_await irods::s3::actions::handle_headobject(socket, parser, url);
+                co_return;
+            }
+            break;
+            /*
+        case boost::beast::http::verb::delete_:
+            // DeleteObject
+            std::cout << "Deleteobject detected" << std::endl;
+            break;
+            */
+        default:
+            std::cerr << "Oh no..." << std::endl;
+            exit(37);
```

Comment:
Is calling `exit` the best we can do here?

Reply:
I suppose it doesn't really benefit from crashing here. The software should be able to continue handling requests even in this case, so I'll just snip that out.l